### DISCUSSION TITLE
Not sure how to use pagination

### DISCUSSION THREAD

**Discussion by indahud** [supabase/supabase#1223] [Questions] [answered]
While going through the docs I found that `range` is responsible for pagination. But I am not quite sure how it works. I am trying to pass range in the query. I am using it with NextJS `getServerSideProps`. I am trying to display 10 records with Next and Prev Buttons. Looking for some help. Thank you in advance. 

```JavaScript
function Cars({ data, count, page }) {
  const router = useRouter();
  return (
    <div>
      {data.map((carsdata) => (
        <ul key={carsdata.id}>
          <li>{carsdata.name}</li>
        </ul>
      ))}

      <button onClick={() => router.push(`/Cars?page=${page - 1}`)}>
        Prev
      </button>
      <button onClick={() => router.push(`/Cars?page=${page + 1}&range=0-9`)}>
        Next
      </button>
    </div>
  );
}

export async function getServerSideProps({ query: { page = 1 } }) {
  const { data, count } = await await supabase
    .from("cars")
    .select("*", { count: "exact" })
    .order("id", { ascending: true })
    .range(0, 9);
    
  return {
    props: {
      data: data,
      count: count,
      page: +page,
    },
  };
}

export default Cars;
```

**Comment by silentworks** [comment, marked as answer]
You just need a function that can calculate the range for you. I have a function I created which works really well and you can use it in your code.

```js
export const getPagination = (page, size) => {
  const limit = size ? +size : 3;
  const from = page ? page * limit : 0;
  const to = page ? from + size : size;

  return { from, to };
};
```

You would add this before your supabase query and use the returned `from` and `to` as your `.range` properties.

```js
export async function getServerSideProps({ query: { page = 1 } }) {
  const { from, to } = getPagination(page, 10);
  const { data, count } = await supabase
    .from("cars")
    .select("*", { count: "exact" })
    .order("id", { ascending: true })
    .range(from, to);
    
  return {
    props: {
      data: data,
      count: count,
      page: +page,
    },
  };
}
```

**Reply by indahud** [reply]
Oh man Thank You very much. Appreciate your help.

**Reply by ghost** [reply]
It looks like `getPagination` returns `size + 1` results.

Here is what I did to fix it:
```jsx
const getPagination = (page, size) => {
  const limit = size ? +size : 3
  const from = page ? page * limit : 0
  const to = page ? from + size - 1 : size - 1

  return { from, to }
}
```

**Reply by pryme0** [reply]
> It looks like `getPagination` returns `size + 1` results.
> 
> Here is what I did to fix it:
> 
> ```js
> const getPagination = (page, size) => {
>   const limit = size ? +size : 3
>   const from = page ? page * limit : 0
>   const to = page ? from + size - 1 : size - 1
> 
>   return { from, to }
> }
> ```

thanks ....this worked for me

**Reply by roker15** [reply]
What if Id is not in sequence or some ids are missing . For example 1,2,8,16 .... are ids
